sound/soc/sof/intel/hda-dsp.c

```diff
@@ -336,6 +337,11 @@ int hda_dsp_set_power_state(struct snd_sof_dev *sdev,
 	dev_vdbg(bus->dev, "D0I3C updated, register = 0x%x\n",
 		 snd_hdac_chip_readb(bus, VS_D0I3C));
 
+	if (d0_substate == SOF_DSP_D0I3)
+		*flags = SOF_PM_PG_STREAMING;
+	else
+		*flags = SOF_PM_PPG | SOF_PM_PCG;
```

Comment:
I don't get this part.

The platform-specific set_power_state returns flags that are then send by an IPC handled in the generic core.

Why can't we just send the IPC here, and leave the core alone?

Reply:
Good point. I tried to make the IPC assembling part generic, but now I agree it is good idea to leave them in platform-specific, will change.